keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ArtifactRepository.kt

```diff
@@ -65,6 +67,19 @@ interface ArtifactRepository {
   )
 }
 
+private val VERSION_COMPARATOR: Comparator<String> = object : Comparator<String> {
+  override fun compare(s1: String, s2: String): Int {
+    return debComparator.compare(s1.toVersion(), s2.toVersion())
+  }
+
+  private val debComparator = DebianVersionComparator()
+
+  private fun String.toVersion() = AppVersion.parseName(this).version
```

Comment:
if its like everything else in Frigga, AppVersion.parseName probably thoroughly enjoys returning null

I think the usage in compare needs to guard against null args (and maybe support null args) to honour the Comparator contract? unless this is some kotlin take on Comparator that sorts out null already

Reply:
Yeah, that's why I had to update the `ArtifactControllerTests` to use valid-ish appversion strings. Guarding against this is a thoroughly good idea, though.